### DISCUSSION TITLE
Colour calibration for SK6812 COLD

### DISCUSSION THREAD

**Discussion by tavalin** [awawa-dev/HyperHDR#544] [Q&A]
Hi @awawa-dev and community.  

I have bought some strips of BTF SK6812 COLD LEDs and so far it's all working fine with the default settings (in LED Hardware Calibration and Image Processing) except the colours need calibrating against my TV.  Is there a tried and tested method for this?  I have attempted to do this by using Red, Green, Blue, White and Grey backgrounds and fiddling with the settings in Image Processing tab but I'm doing it very unscientifically and I'm not sure what settings are meant to fix my specific problems.

Problems so far:

1) The white seems to bleed through so that Red (255,0,0) ends up looking a bit orange/pink - is dialing back the white channel brightness in LED Hardware tab the right way to fix this?

2) The green doesn't get green enough.  I have attempted to fix this by reducing the Blue Temperature to about 10 - works to fix the green but then this throws off the white.

3) At this point I couldn't find a satisfactory solution to get all colours properly calibration and then I posted here for help 😄 

Edit: Is there any description of what the temperature settings are meant to achieve? There doesn't seem to be a descriptive help text for them and they seem to noticeable affect brightness and power consumption 

Thanks in advance

**Comment by awawa-dev** [comment]
Hi
Which grabber do you use?

**Reply by tavalin** [reply]
I am using Piccap with flatbuffers on my LG TV.

**Reply by awawa-dev** [reply]
I see. The problem is not with the calibration of the LED, but with the video source. Unfortunately, the webos capturing that Piccap uses is partially broken and can only be fixed to some extent by HyperHDR. For best effect for your TV, you need to auto-calibrate LUT using another HyperHDR instance on your PC redirecting piccap there temporarily (currently it won't work on the TV itself because webos build disables USB grabber supports, patch in underway https://github.com/awawa-dev/HyperHDR/pull/518). It's described here: https://www.hyperhdr.eu/2022/04/usb-grabbers-hdr-to-sdr-quality-test.html#chapter2. Even if calibration is applied, the calibration's error related to quality will usually be in the range of 24000-26000 points, by comparison a USB grabber may have it below 500 (the lower the error the better) even when HDR10 signal is used. The results can be found in the logs just after the calibration finishes.

**Reply by tavalin** [reply]
Thanks for your reply.  Just to clarify and hopefully make sure I understand the LUT calibration process....I am not running HyperHDR on my LG TV.  I am running HyperHDR on a separate server (because I couldn't get it running on my LG TV and am a bit concerned about the performance) that receives the data from Piccap, and is attached via USB to an ESP32 running HyperSerial32.  Can I do the calibration like this?

**Reply by awawa-dev** [reply]
Yes, you can calibrate it on your server. Though I'm not sure if USB grabber needs to be present in the system even if you're using a flatbuffers as a video source. Anyway #518 should fix it also.

**Reply by tavalin** [reply]
Is there a way to get the binary for https://github.com/awawa-dev/HyperHDR/pull/518? The GitHub actions doesn't seem to have it linked

**Reply by awawa-dev** [reply]
Yes, it's being built right now https://github.com/awawa-dev/HyperHDR/actions/runs/4609721188

**Reply by tavalin** [reply]
I have run the version linked above and tried to run the calibration and received the following in my log:

```
2023-04-05T12:28:02.864Z [CALIBRATOR] (LutCalibrator.cpp:1244) ----------------- Preparing and saving LUT table --------------------
2023-04-05T12:28:02.864Z [CALIBRATOR] (LutCalibrator.cpp:1245) Initial mode: YES
2023-04-05T12:28:02.864Z [CALIBRATOR] (LutCalibrator.cpp:1246) Using YUV coefs: FCC
2023-04-05T12:28:02.864Z [CALIBRATOR] (LutCalibrator.cpp:1247) YUV table range: FULL
2023-04-05T12:28:02.892Z [CALIBRATOR] (LutCalibrator.cpp:1258) LUT RGB HDR table (1/3) is ready
2023-04-05T12:28:02.921Z [CALIBRATOR] (LutCalibrator.cpp:1296) LUT YUV HDR table (2/3) is ready
2023-04-05T12:28:03.130Z [CALIBRATOR] (LutCalibrator.cpp:1327) LUT YUV table (3/3) is ready
2023-04-05T12:28:03.187Z [CALIBRATOR] (LutCalibrator.cpp:1331) Your new LUT file is saved as /root/.hyperhdr/lut_lin_tables.3d. Is ready for usage: NO. It's temporary LUT table without HDR information..
2023-04-05T12:28:03.187Z [CALIBRATOR] (LutCalibrator.cpp:1332) ---------------------- LUT table is saved -----------------------
2023-04-05T12:28:03.187Z [CALIBRATOR] (LutCalibrator.cpp:1333) 
2023-04-05T12:28:03.193Z [V4L2:AUTO] Could not find any capture device 
2023-04-05T12:28:04.797Z [CALIBRATOR] (LutCalibrator.cpp:182) Using flatbuffers/protobuffers as a source
```

Is this complete? I didn't see anything like you mentioned that showed a score.

**Reply by awawa-dev** [reply]
No it's no completed - it stops and waiting for the valid video input. Make sure you are receiving full "mosaic" (including encoded status lines at the top and the bottom) in the HyperHDR video preview: for some resolutions webos capturing may cut off some lines.

**Reply by awawa-dev** [reply]
Also for calibration you must use high resolution mode in piccap.
